save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/basic/organizations/OrganizationContestsMenu.kt

```diff
@@ -112,7 +139,60 @@ external interface OrganizationContestsTableProps<D : Any> : TableProps<D> {
 private fun organizationContestsMenu() = FC<OrganizationContestsMenuProps> { props ->
     val (isToUpdateTable, setIsToUpdateTable) = useState(false)
     val (isContestCreationModalOpen, setIsContestCreationModalOpen) = useState(false)
+    val (isDeleteContestModalOpen, setDeleteContestModalOpen) = useState(false)
+    val (deletingContest, setDeletingContest) = useState(ContestDto.empty)
+    val (confirmLabel, setConfirmLabel) = useState("")
+    val (confirmMessage, setConfirmMessage) = useState("")
+    val (contests, setContests) = useState<Set<ContestDto>>(setOf())
     val refreshTable = { setIsToUpdateTable { !it } }
+    val deleteContestFun = useDeferredRequest {
+        val deleteContest = deletingContest.copy(status = ContestStatus.DELETED)
+        val response = post(
+            "$apiUrl/contests/update",
+            jsonHeaders,
+            Json.encodeToString(deleteContest),
+            loadingHandler = ::noopLoadingHandler,
+        )
+        if (response.ok) {
+            setContests(contests.minusElement(deletingContest))
+            refreshTable()
+        }
+    }
+
+    useRequest {
+        val response = get(
+            url = "$apiUrl/contests/by-organization?organizationName=${props.organizationName}",
+            headers = jsonHeaders,
+            loadingHandler = ::loadingHandler,
+        )
+        val newContests = if (response.ok) {
+            response.unsafeMap {
+                it.decodeFromJsonString<List<ContestDto>>()
+            }
+                .toSet()
+        } else {
+            setOf()
+        }
+        setContests(newContests)
+        refreshTable()
+    }
+
+    displayModal(
```

Comment:
suggestion to add extension for `displayModal` and re-use:
```suggestion
 fun ChildrenBuilder.displayConfirmationModal(
    windowOpenness: WindowOpenness,
    title: String,
    message: String,
    modalStyle: Styles = mediumTransparentModalStyle,
    successAction: () -> Unit,
) {
    displayModal(
        isOpen = windowOpenness.isOpen(),
        title = title,
        message = message,
        modalStyle = modalStyle,
        onCloseButtonPressed = windowOpenness.closeWindowAction()
    ) {
        buttonBuilder("Ok") {
            successAction()
            windowOpenness.closeWindow()
        }
        buttonBuilder("Cancel", "secondary") {
            windowOpenness.closeWindow()
        }
    }
}
```

Reply:
What for? This is the same DisplayModal but with a different name. Then it will have to be replaced throughout the project.

Reply:
I don't think it's necessary for this task. You can start a new issue.

Reply:
> What for? 

to avoid creating buttons in each place where confirmation window is required


> Then it will have to be replaced throughout the project.

No need to replace in all places, but you can create a new method and re-use it in your PR

Reply:
Fixed